migration/v2.27/README.md

```diff
@@ -0,0 +1,150 @@
+# Upgrade ${old_version} to ${new_version}
+
+## Prerequisites
+
+- [ ] Notify the users (if any) before the upgrade starts;
+- [ ] Check if there are any pending changes to the environment;
+- [ ] Check the state of the environment, pods, nodes and backup jobs:
+
+    ```bash
+    ./compliantkubernetes-apps/bin/ck8s test sc|wc
+    ./compliantkubernetes-apps/bin/ck8s ops kubectl sc|wc get pods -A -o custom-columns=NAMESPACE:metadata.namespace,POD:metadata.name,READY-false:status.containerStatuses[*].ready,REASON:status.containerStatuses[*].state.terminated.reason | grep false | grep -v Completed
+    ./compliantkubernetes-apps/bin/ck8s ops kubectl sc|wc get nodes
+    ./compliantkubernetes-apps/bin/ck8s ops kubectl sc|wc get jobs -A
+    velero get backup
+    ```
+
+- [ ] Silence the notifications for the alerts. e.g you can use [alertmanager silences](https://prometheus.io/docs/alerting/latest/alertmanager/#silences);
+
+## Steps that can be done before the upgrade - non-disruptive
+
+1. Checkout the new release: `git switch -d ${full_version}`
+
+1. Switch to the correct remote: `git submodule sync`
+
+1. Update the kubespray submodule: `git submodule update --init --recursive`
+
+1. Run `bin/ck8s-kubespray upgrade both ${new_version} prepare` to update your config.
+
+    > [!NOTE]
+    > It is possible to update `wc` and `sc` config separately by replacing `both` when running the `upgrade` command, e.g. the following will only update config for the workload cluster:
+    >
+    > ```bash
+    > bin/ck8s-kubespray upgrade wc ${new_version} prepare
+    > ```
+
+1. Download the required files on the nodes
+
+    ```bash
+    ./bin/ck8s-kubespray run-playbook sc upgrade_cluster.yml -b --tags=download
+    ./bin/ck8s-kubespray run-playbook wc upgrade_cluster.yml -b --tags=download
+    ```
+
+## Upgrade steps
+
+These steps will cause disruptions in the environment.
+
+1. For UpCloud environments, update loadbalancer configuration
```

Comment:
Will this cause disruptions in the environment to update the terraform state for the LBs?

Reply:
Nope, without this migration it would be disruptive (LB recreation) But this fixes that so that it's trasparent

Reply:
Okay, so it can be moved to the non-disruptive section?

Reply:
https://github.com/elastisys/compliantkubernetes-kubespray/pull/411/commits/a3b38d41b7fe13903018303b409450ce2f6a52ba